registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -171,171 +168,151 @@ NormalDistributionsTransform<PointSource, PointTarget>::computeTransformation (P
 
   // Store transformation probability.  The realtive differences within each scan registration are accurate
   // but the normalization constants need to be modified for it to be globally accurate
-  trans_probability_ = score / static_cast<double> (input_->points.size ());
+  trans_probability_ = score / static_cast<double> (input_->size ());
 }
 
 
 template<typename PointSource, typename PointTarget> double
 NormalDistributionsTransform<PointSource, PointTarget>::computeDerivatives (Eigen::Matrix<double, 6, 1> &score_gradient,
                                                                             Eigen::Matrix<double, 6, 6> &hessian,
                                                                             PointCloudSource &trans_cloud,
-                                                                            Eigen::Matrix<double, 6, 1> &p,
+                                                                            Eigen::Matrix<double, 6, 1> &transform,
                                                                             bool compute_hessian)
 {
-  // Original Point and Transformed Point
-  PointSource x_pt, x_trans_pt;
-  // Original Point and Transformed Point (for math)
-  Eigen::Vector3d x, x_trans;
-  // Occupied Voxel
-  TargetGridLeafConstPtr cell;
-  // Inverse Covariance of Occupied Voxel
-  Eigen::Matrix3d c_inv;
-
   score_gradient.setZero ();
   hessian.setZero ();
   double score = 0;
 
   // Precompute Angular Derivatives (eq. 6.19 and 6.21)[Magnusson 2009]
-  computeAngleDerivatives (p);
+  computeAngleDerivatives (transform);
 
   // Update gradient and hessian for each point, line 17 in Algorithm 2 [Magnusson 2009]
-  for (std::size_t idx = 0; idx < input_->points.size (); idx++)
+  for (std::size_t idx = 0; idx < input_->size (); idx++)
   {
-    x_trans_pt = trans_cloud[idx];
+  // Transformed Point
+    const auto& x_trans_pt = trans_cloud[idx];
+    Eigen::Vector3d x_trans = x_trans_pt.getVector3fMap().template cast<double>();
 
-    // Find nieghbors (Radius search has been experimentally faster than direct neighbor checking.
+    // Find neighbors (Radius search has been experimentally faster than direct neighbor checking.
     std::vector<TargetGridLeafConstPtr> neighborhood;
     std::vector<float> distances;
     target_cells_.radiusSearch (x_trans_pt, resolution_, neighborhood, distances);
 
     for (typename std::vector<TargetGridLeafConstPtr>::iterator neighborhood_it = neighborhood.begin (); neighborhood_it != neighborhood.end (); ++neighborhood_it)
```

Comment:
for range loop

Reply:
Done.
I also replaced the for at L390 in the same way.